flytekit/bin/entrypoint.py

```diff
@@ -376,6 +377,8 @@ def _execute_task(
         dynamic_addl_distro,
         dynamic_dest_dir,
     ) as ctx:
+        if all(os.path.realpath(path) != dynamic_dest_dir for path in sys.path):
+            sys.path.append(dynamic_dest_dir)
```

Comment:
I think we should add [dest_dir](https://github.com/flyteorg/flytekit/blob/bc2e000cc8d710ed3d135cdbf3cbf257c5da8100/flytekit/bin/entrypoint.py#L532) to PYTHONPATH instead of `dynamic_dest_dir`